src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -7948,17 +7948,34 @@ public void CS0501ERR_ConcreteMissingBody02()
         {
             var text = @"abstract class C
 {
-    public int P { get; set { } }
-    public int Q { get { return 0; } set; }
+    public int P { get; set { } } // PROTOTYPE(semi-auto-props): Should be an error prior to semi-auto-props. ie, uncomment the Diagnostic(...) line in VerifyDiagnostics.
+    public int Q { get { return 0; } set; } // PROTOTYPE(semi-auto-props): Should be an error prior to semi-auto-props. ie, uncomment the Diagnostic(...) line in VerifyDiagnostics.
     public extern object R { get; } // no error
     protected abstract object S { set; } // no error
 }
 ";
-            CreateCompilation(text).VerifyDiagnostics(
+            CreateCompilation(text, parseOptions: TestOptions.Regular10).VerifyDiagnostics(
                 // (3,20): error CS0501: 'C.P.get' must declare a body because it is not marked abstract, extern, or partial
-                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "get").WithArguments("C.P.get"),
+                //Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "get").WithArguments("C.P.get"),
                 // (4,38): error CS0501: 'C.Q.set' must declare a body because it is not marked abstract, extern, or partial
-                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "set").WithArguments("C.Q.set"),
+                //Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "set").WithArguments("C.Q.set"),
+                // (5,30): warning CS0626: Method, operator, or accessor 'C.R.get' is marked external and has no attributes on it. Consider adding a DllImport attribute to specify the external implementation.
+                Diagnostic(ErrorCode.WRN_ExternMethodNoImplementation, "get").WithArguments("C.R.get"));
+        }
+
+        [Fact]
+        public void CS0501ERR_ConcreteMissingBody02_SemiAutoProps()
+        {
+            var text = @"abstract class C
+{
+    public int P { get; set { } }
```

Comment:
Per my understanding to the [updated spec](https://github.com/dotnet/csharplang/blob/main/proposals/semi-auto-properties.md#specification-changes), this line is going to be an auto implemented property (ie has no errors) (probably a warning for the unused synthesized field). Is my understanding correct? Should I then have 2 "IsAutoProperty" methods, one that's used for C# 11 and one that's used for C# 10? Or is it okay to have the new logic for all language versions?

cc @CyrusNajmabadi

Reply:
>Is my understanding correct? 

Yes. That's correct.

Reply:
> Or is it okay to have the new logic for all language versions?

Imo, it would be fine to have the new logic for an language versions.